src/components/internals/localized-input-toggle/localized-input-toggle.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import FlatButton from '@commercetools-uikit/flat-button';
+import { WorldIcon } from '@commercetools-uikit/icons';
+import messages from '../messages/localized-input/messages';
+
+const LocalizedInputToggle = (props) => {
+  const intl = useIntl();
+
+  const labelMessage = props.isOpen ? props.hideMessage : props.showMessage;
+  const label =
+    typeof labelMessage === 'string'
+      ? labelMessage
+      : intl.formatMessage(labelMessage, {
+          remainingLanguages: props.remainingLocalizations,
+        });
+
+  return (
+    <FlatButton
+      icon={props.icon}
+      label={label}
+      onClick={props.onClick}
+      isDisabled={props.isDisabled}
+    />
+  );
+};
+
+const intlMessageShape = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  description: PropTypes.string,
+  defaultMessage: PropTypes.string.isRequired,
+});
+
+LocalizedInputToggle.displayName = 'LocalizedInputToggle';
+LocalizedInputToggle.propTypes = {
+  icon: PropTypes.node.isRequired,
+  isOpen: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  isDisabled: PropTypes.bool,
+  showMessage: PropTypes.oneOfType([PropTypes.string, intlMessageShape])
```

Comment:
We allow passing the pre-translated string because it's useful for the LocalizedMoneyInput (because the variable is for "currencies" instead of "languages). All the other Localized inputs can keep the default.